sdks/java/core/src/test/java/org/apache/beam/sdk/util/IOChannelUtilsTest.java

```diff
@@ -80,6 +80,13 @@ public void testLargeShardCount() {
   }
 
   @Test
+  public void testFilePrefix() throws Exception {
+    IOChannelUtils.getFactory("file://tmp");
+    IOChannelUtils.getFactory("file:/tmp");
+    IOChannelUtils.getFactory("file:tmp");
```

Comment:
can you add some tests of creating temp files and then actually reading them back? Based on a discussion with @tgroh, I suspect that there may be some underlying issues where the `FileIOChannelFactory` does not gracefully handle the switch from Paths (which it expects now) to URIs.